File: tools/ghidra_x360/Validate.java
// Ghidra Java script: validate the recovered Xbox PPC PE loads + disassembles.
import ghidra.app.script.GhidraScript;
import ghidra.program.model.address.Address;
import ghidra.program.model.mem.MemoryBlock;
import ghidra.program.model.listing.Instruction;
import ghidra.app.cmd.disassemble.DisassembleCommand;

public class Validate extends GhidraScript {
    public void run() throws Exception {
        println("== VALIDATE ==");
        println("LANG: " + currentProgram.getLanguageID());
        println("IMAGEBASE: " + currentProgram.getImageBase());
        for (MemoryBlock b : currentProgram.getMemory().getBlocks()) {
            println("BLOCK " + b.getName() + " " + b.getStart() + " - " + b.getEnd()
                    + " (" + b.getSize() + ") x=" + b.isExecute());
        }
        Address addr = currentProgram.getAddressFactory().getAddress("0x82170000");
        new DisassembleCommand(addr, null, true).applyTo(currentProgram);
        println("DISASM @0x82170000 (first .pdata function):");
        Address a = addr;
        for (int i = 0; i < 16; i++) {
            Instruction ins = currentProgram.getListing().getInstructionAt(a);
            if (ins == null) { println("  " + a + ": (no instruction)"); break; }
            println("  " + a + ": " + ins.toString());
            a = ins.getMaxAddress().add(1);
        }
        println("== END VALIDATE ==");
    }
}

File: tools/ghidra_x360/validate.py
# Ghidra Jython — validate that the recovered Xbox PPC PE loads + disassembles.
# Prints language, image base, memory blocks, and disassembles at the first
# .pdata function (0x82170000) to confirm the language decodes real PPC.
prog = currentProgram
print("== VALIDATE ==")
print("LANG: %s" % prog.getLanguageID())
print("IMAGEBASE: %s" % prog.getImageBase())
print("BLOCKS:")
for b in prog.getMemory().getBlocks():
    print("  %-10s %s - %s  (%d bytes) x=%s" % (b.getName(), b.getStart(), b.getEnd(),
          b.getSize(), b.isExecute()))

from ghidra.app.cmd.disassemble import DisassembleCommand
af = prog.getAddressFactory()
addr = af.getAddress("0x82170000")
DisassembleCommand(addr, None, True).applyTo(prog)
listing = prog.getListing()
print("DISASM @0x82170000 (first .pdata function):")
a = addr
for i in range(16):
    ins = listing.getInstructionAt(a)
    if ins is None:
        print("  %s: (no instruction)" % a); break
    print("  %s: %s" % (a, ins))
    a = ins.getMaxAddress().add(1)
print("== END VALIDATE ==")
